tests/unit/src/test/scala/tests/DebugProtocolSuite.scala

```diff
@@ -106,4 +110,66 @@ class DebugProtocolSuite extends BaseLspSuite("debug-protocol") {
       output <- debugger.allOutput
     } yield assertNoDiff(output, "Bar\n")
   }
+
+  test("run-unresolved-params") {
+    for {
+      _ <- server.initialize(
+        s"""/metals.json
+           |{
+           |  "a": {}
+           |}
+           |/a/src/main/scala/a/Main.scala
+           |package a
+           |object Main {
+           |  def main(args: Array[String]) = {
+           |    print(args(0))
+           |  }
+           |}
+           |""".stripMargin
+      )
+      _ = Thread.sleep(1000) //it fails otherwise
```

Comment:
Could you elaborate on the failure cause?

Reply:
Also, this is a place where possible flakiness may come from due to some underlying race conditions.